projects/ponder/index.js

```diff
@@ -0,0 +1,192 @@
+const factoryAbi = 'function allPairsLength() external view returns (uint256)';
+const allPairsAbi = 'function allPairs(uint256) external view returns (address)';
+const pairAbis = {
+  token0: 'function token0() external view returns (address)',
+  token1: 'function token1() external view returns (address)',
+  getReserves: 'function getReserves() external view returns (uint112 reserve0, uint112 reserve1, uint32 blockTimestampLast)'
+};
+
+// Chunk size for processing large number of pairs
+const CHUNK_SIZE = 100;
+
+async function tvl(api) {
+  try {
+    // Get total number of pairs
+    const allPairsLength = await api.call({
+      abi: factoryAbi,
+      target: addresses.Factory,
+    });
+    
+    if (Number(allPairsLength) === 0) {
+      return {}; // No pairs found
+    }
+    
+    const totalPairs = Number(allPairsLength);
+    
+    // Process pairs in chunks to avoid timeouts and memory issues
+    for (let chunkStart = 0; chunkStart < totalPairs; chunkStart += CHUNK_SIZE) {
+      const chunkEnd = Math.min(chunkStart + CHUNK_SIZE, totalPairs);
+      
+      // Get pair addresses for this chunk
+      const pairCalls = Array.from(
+        { length: chunkEnd - chunkStart }, 
+        (_, i) => ({
+          target: addresses.Factory,
+          params: [chunkStart + i]
+        })
+      );
+      
+      const pairAddresses = await api.multiCall({
+        abi: allPairsAbi,
+        calls: pairCalls
+      });
+      
+      // Get token addresses and reserves for this chunk
+      const [token0Addresses, token1Addresses, reserves] = await Promise.all([
+        api.multiCall({
+          abi: pairAbis.token0,
+          calls: pairAddresses.map(p => ({ target: p }))
+        }),
+        api.multiCall({
+          abi: pairAbis.token1,
+          calls: pairAddresses.map(p => ({ target: p }))
+        }),
+        api.multiCall({
+          abi: pairAbis.getReserves,
+          calls: pairAddresses.map(p => ({ target: p }))
+        })
+      ]);
+      
+      // Add balances for this chunk
+      for (let i = 0; i < pairAddresses.length; i++) {
+        const token0 = token0Addresses[i];
+        const token1 = token1Addresses[i];
+        const [reserve0, reserve1] = reserves[i];
+        
+        api.add(token0, reserve0);
+        api.add(token1, reserve1);
+      }
+    }
+  } catch (error) {
+    console.error(`Error in Ponder TVL calculation: ${error}`);
+    return {};
+  }
+}
+
+async function staking(api) {
+  try {
+    // Get PONDER balance in staking contract
+    const stakedPonderBalance = await api.call({
+      abi: 'erc20:balanceOf',
+      target: addresses.PonderToken,
+      params: [addresses.PonderStaking]
+    });
+    
+    api.add(addresses.PonderToken, stakedPonderBalance);
+  } catch (error) {
+    console.error(`Error in Ponder staking TVL calculation: ${error}`);
+    return {};
+  }
+}
+
+async function pool2(api) {
+  try {
+    const ponderTokenAddress = addresses.PonderToken.toLowerCase();
```

Comment:
if these pool2 pools are already included in the core tvl function then they cant also be part of pool2 tvl